src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1485,6 +1527,8 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                             semicolon);
 
                     case SyntaxKind.InterfaceKeyword:
+                        RoslynDebug.Assert(openBrace != null);
```

Comment:
>RoslynDebug [](start = 24, length = 11)

I don't see how this assertion is correct.  It appears that the above code would accept `interface A;`  I think we either need to make the curly braces optional or make missing tokens for them with an error.

Reply:
Similarly this code appears to be dropping the parameter list on the floor.  I think it needs to be either added to the syntax, not permitted by the parser, or you need to build a class declaration when that happens.

---
In reply to: [361761780](https://github.com/dotnet/roslyn/pull/40467#discussion_r361761780) [](ancestors = 361761780)